astropy/units/tests/test_quantity.py

```diff
@@ -1606,3 +1608,97 @@ def test_masked_quantity_str_repr():
                                   mask=[True, False, True, False])
     str(masked_quantity)
     repr(masked_quantity)
+
+
+class TestQuantityTyping:
```

Comment:
I'd move this to a typing specific test function.

Reply:
As in a different file, like `test_quantity_typing`?

Reply:
Sorry, yes, I meant a different test file...